substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/phases/NativeImageInlineDuringParsingPlugin.java

```diff
@@ -0,0 +1,180 @@
+        }
+
+        if (method.getAnnotation(NeverInline.class) != null || method.getAnnotation(NeverInlineTrivial.class) != null) {
+            return null;
+        }
+
+        if (method.getAnnotation(RestrictHeapAccess.class) != null || method.getAnnotation(Uninterruptible.class) != null ||
+                b.getMethod().getAnnotation(RestrictHeapAccess.class) != null || b.getMethod().getAnnotation(Uninterruptible.class) != null) {
+            /*
+             * Caller or callee have an annotation that might prevent inlining. We don't check the
+             * exact condition but instead always bail out for simplicity.
+             */
+            return null;
+        }
+
+        if (!method.hasBytecodes())
+            return null;
+
+        if (method.isSynchronized())
+          return null;
+
+        if (providers.getGraphBuilderPlugins().getInvocationPlugins().lookupInvocation(method) != null)
+            return null;
+
+        if (((AnalysisMethod)method).buildGraph(b.getDebug(), method, providers, Purpose.ANALYSIS) != null)
+            /* Method has a manually constructed graph via GraphProvider. */
+            return null;
+
+        // method is declared in user's class
+        if(SubstrateOptions.Class.getValue().equals(method.format("%H"))) {
+            // b has info for caller
+            int nodeCountCaller = b.getGraph().getNodeCount();
+            // get graph for callee
+            StructuredGraph graph = new StructuredGraph.Builder(b.getOptions(), b.getDebug()).method(method).build();
+
+            AnalysisGraphBuilderPhase graphbuilder = new AnalysisGraphBuilderPhase(providers, ((SharedGraphBuilderPhase.SharedBytecodeParser) b).getGraphBuilderConfig(), OptimisticOptimizations.NONE, null, providers.getWordTypes());
+            graphbuilder.apply(graph);
+
+            int nodeCountCallee = graph.getNodeCount();
+
+            System.out.println(b.getMethod().format("Caller: %n (class: %H), par: %p, ")
+                        + "node count: " + nodeCountCaller
+                        + method.format("\nCallee: %n (class: %H), par: %p, ")
+                        + "node count: " + nodeCountCallee);
+            // first, look for getters, setters and similar methods
+            for (Node node : graph.getNodes()) {
+                System.out.print(node.toString());
+                if (node instanceof LoadFieldNode)
+                    System.out.print(" - node represents a read of a static or instance field.");
+                if (node instanceof StoreFieldNode)
+                    System.out.print(" - node represents a write to a static or instance field.");
+                if (node instanceof ParameterNode)
+                    System.out.print(" - node represents a placeholder for an incoming argument to a function call.");
+                if (node instanceof ConstantNode)
+                    System.out.print(" - node represents a constant");
+                System.out.println();
```

Comment:
@vjovanov
uspesno se parsira graf za metod, za sada su samo ispisani cvorovi

Reply:
That is awesome. Now, we can start detecting the simplest methods.